packages/flutter/lib/src/material/text_theme.dart

```diff
@@ -78,66 +81,197 @@ class TextTheme with Diagnosticable {
   ///
   /// If you do decide to create your own text theme, consider using one of
   /// those predefined themes as a starting point for [copyWith] or [apply].
+  ///
+  /// Please note that you can not mix and match the 2018 styles with the 2021
+  /// styles. Only one or the other is allowed in this constructor. The 2018
+  /// styles will be deprecated and removed eventually.
   const TextTheme({
-    this.headline1,
-    this.headline2,
-    this.headline3,
-    this.headline4,
-    this.headline5,
-    this.headline6,
-    this.subtitle1,
-    this.subtitle2,
-    this.bodyText1,
-    this.bodyText2,
-    this.caption,
-    this.button,
-    this.overline,
-  });
+    TextStyle? displayLarge,
+    TextStyle? displayMedium,
+    TextStyle? displaySmall,
+    this.headlineLarge,
+    TextStyle? headlineMedium,
+    TextStyle? headlineSmall,
+    TextStyle? titleLarge,
+    TextStyle? titleMedium,
+    TextStyle? titleSmall,
+    TextStyle? bodyLarge,
+    TextStyle? bodyMedium,
+    TextStyle? bodySmall,
+    TextStyle? labelLarge,
+    this.labelMedium,
+    TextStyle? labelSmall,
+    TextStyle? headline1,
+    TextStyle? headline2,
+    TextStyle? headline3,
+    TextStyle? headline4,
+    TextStyle? headline5,
+    TextStyle? headline6,
+    TextStyle? subtitle1,
+    TextStyle? subtitle2,
+    TextStyle? bodyText1,
+    TextStyle? bodyText2,
+    TextStyle? caption,
+    TextStyle? button,
+    TextStyle? overline,
+  }) : assert(
+         (displayLarge == null && displayMedium == null && displaySmall == null && headlineMedium == null &&
+             headlineSmall == null && titleLarge == null && titleMedium == null && titleSmall == null &&
+             bodyLarge == null && bodyMedium == null && bodySmall == null && labelLarge == null && labelSmall == null) ||
+         (headline1 == null && headline2 == null && headline3 == null && headline4 == null &&
+             headline5 == null && headline6 == null && subtitle1 == null && subtitle2 == null &&
+             bodyText1 == null && bodyText2 == null && caption == null && button == null && overline == null),
+         'Cannot mix 2018 and 2021 terms in call to TextTheme() constructor.'
+       ),
+       displayLarge = displayLarge ?? headline1,
+       displayMedium = displayMedium ?? headline2,
+       displaySmall = displaySmall ?? headline3,
+       headlineMedium = headlineMedium ?? headline4,
+       headlineSmall = headlineSmall ?? headline5,
+       titleLarge = titleLarge ?? headline6,
+       titleMedium = titleMedium ?? subtitle1,
+       titleSmall = titleSmall ?? subtitle2,
+       bodyLarge = bodyLarge ?? bodyText1,
+       bodyMedium = bodyMedium ?? bodyText2,
+       bodySmall = bodySmall ?? caption,
+       labelLarge = labelLarge ?? button,
+       labelSmall = labelSmall ?? overline;
+
+  /// Large size of the display styles.
```

Comment:
Are we really helping anyone with comments like this?  There must be a reason for calling some of the fonts "display". Presumably it has something to do with using them for short standalone text that's not a title (whatever that is). As far as the "large" qualifier goes, it would be worth characterizing this TextStyle as the _largest_.

Here and below.

Reply:
Hmm yeah its tricky to describe these styles, I copied the descriptions from https://m3.material.io/styles/typography/applying-scaling-type for each category (and put it in the 2nd sentence of each comment) and then I updated to use largest, smallest, etc. Let me know what you think?

Reply:
The descriptions look good.